Neo4j.Driver/Neo4j.Driver/Internal/Result/ResultCursorBuilder.cs

```diff
@@ -139,7 +144,11 @@ public async ValueTask<IResultSummary> ConsumeAsync()
     public void RunCompleted(long queryId, string[] fields, IResponsePipelineError error)
     {
         _queryId = queryId;
-        _fields = fields;
+
+        _fieldLookup = fields.Select((field, index) => (field, index))
```

Comment:
can we do this the old fashioned way, and do it in 1 pass.
```
var invariant = new Dictionary<string, int>(fields.Count, StringComparer.InvariantCultureIgnoreCase);
var exact = new Dictionary<string, int>(fields.Count);
for (var i = 0; i < fields.Count; i++)
{
    invariant.Add(fields[i], i);
    exact.Add(fields[i], i);
}
_invariantFieldLookup = invariant;
_fieldLookup = exact;
```

Reply:
Done